agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/components/AgendaOverviewDatePicker.kt

```diff
@@ -0,0 +1,137 @@
+@file:OptIn(ExperimentalMaterial3Api::class)
+
+package com.pronoidsoftware.agenda.presentation.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.DatePickerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.pronoidsoftware.agenda.presentation.R
+import com.pronoidsoftware.core.presentation.designsystem.ArrowDownIcon
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+import com.pronoidsoftware.core.presentation.ui.toLocalDate
+import com.pronoidsoftware.core.presentation.ui.toMillis
+import com.pronoidsoftware.core.presentation.ui.today
+import java.util.Locale
+import kotlinx.datetime.Clock
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun AgendaOverviewDatePicker(
+    selectedDate: LocalDate,
+    onSelectDate: (LocalDate) -> Unit,
+    modifier: Modifier = Modifier,
+    clock: Clock = Clock.System,
+) {
+    var openDialog by remember {
+        mutableStateOf(false)
+    }
+
+    // don't use rememberDatePickerState because we want to be able to set the selected date
+    // from the date widget on the agenda overview screen too
+    val datePickerState = DatePickerState(
+        locale = Locale.getDefault(),
+        initialSelectedDateMillis = selectedDate.toMillis(),
+    )
```

Comment:
This should be put in `remember` right? Otherwise, the state is reset after every recomposition

Reply:
I tried holding onto the date picker state with both remember and moving it up to the agenda overview state, but it became quite tricky to keep the state correctly updated when selecting various dates. The date picker state holds onto its own selected date and updates it each time a user clicks a new date, so that then introduced a new variable to hold onto the originally opened date so that the cancel button would work correctly, but I couldn't get that quite right. Since this implementation works for me, and it helps keep a single source of truth with only the selected date in the agenda overview state and this composable only caring about the date picker state, I'll keep it for now and might revisit if performance is an issue. But I'll add a UI test for this to help ensure it behaves as expected.